### PR TITLE
Add `.before` and `.after` arguments to `mutate`

### PR DESCRIPTION
**Pull request by eutwt**
Example with this PR loaded:

``` r
library(dtplyr)
library(dplyr, warn.conflicts = FALSE)

lazy_dt(data.frame(x = 1, y = 1)) %>% 
  mutate(z = 1, .before = y)
#> Source: local data table [1 x 3]
#> Call:   setcolorder(copy(`_DT1`)[, `:=`(z = 1)], c("x", "z", "y"))
#> 
#>       x     z     y
#>   <dbl> <dbl> <dbl>
#> 1     1     1     1
#> 
#> # Use as.data.table()/as.data.frame()/as_tibble() to access results
```

<sup>Created on 2021-08-29 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

### CODE REVIEW

File: tests/testthat/test-step-mutate.R [hunk 1]
```diff
@@ -105,3 +105,27 @@ test_that("emtpy mutate returns input", {
...
+test_that("can use .before and .after to control column position", {
+  dt <- lazy_dt(data.frame(x = 1, y = 2))
+  expect_named(
+    mutate(dt, z = 1) %>% as_tibble(),
+    c("x", "y", "z")
+  )
+  expect_named(
+    mutate(dt, z = 1, .before = x) %>% as_tibble(),
+    c("z", "x", "y")
+    )
```
Comment (hadley) [inline tests/testthat/test-step-mutate.R:120]: Indenting is slightly off here and below.

**Review by hadley — COMMENTED** [review]
In general, I don't think dtplyr needs to provide experimental arguments, but I'm pretty sure that all the current dplyr experimental arguments are going to stay, so this still makes sense. Otherwise looks good, just one small styling suggestion and it needs a news bullet 😄

File: tests/testthat/test-step-mutate.R (hunk 1 above)
Comment (eutwt) [reply tests/testthat/test-step-mutate.R]: Thanks, fixed.

**Comment by eutwt** [comment]
Should be all set now

**Comment by hadley** [comment]
BTW we add news bullets to the top, since it makes resolving the merge conflicts a little easier.

**Pull request merged**